include/pistache/client.h

```diff
@@ -76,10 +76,19 @@ struct Connection : public std::enable_shared_from_this<Connection> {
     void connect(Address addr);
     void close();
     bool isIdle() const;
-    bool isConnected() const;
```

Comment:
You don't need to remove 'const' to lock a mutex.  Leave const but add the "mutable" attribute to the mutex definition in the class definition.

See:
  https://en.cppreference.com/w/cpp/language/cv
  https://herbsutter.com/2013/05/24/gotw-6a-const-correctness-part-1-3/